### PR TITLE
add optional validation task

### PR DESCRIPTION
**Pull request by jkingyens**
adds a "gulp validate" to invoke the runtime validator on each AMP example. If success output "PASS", otherwise output "FAIL" and show AMP validation errors from the javascript console.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -120,6 +121,12 @@ gulp.task("compile:favicons", function() {
       .pipe(gulp.dest("dist/favicons"));
 });
 
+gulp.task('validate:example', 'validate example html files', function() {
+  return gulp.src('src/*.html')
+    .pipe(compileExample('./src/templates/', 'example.html'))
```
Comment (sebastianbenz) [inline gulpfile.js]: We could avoid duplicating the compileExample step by adding a dependency to the build (and clean) task. For a sample see the deploy task, which does the same.

File: tasks/validate-example.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        child.on('exit', function() {
+          let printedOutput = '';
+          const parsedOutput = JSON.parse(output);
+          const exampleKey = 'http://localhost:30000/' + inputFilename;
+          if (parsedOutput[exampleKey].success) {
+            printedOutput = 'PASS';
+          } else {
+            const errorList = parsedOutput[exampleKey].errors;
+            printedOutput = 'FAIL:\n';
+            errorList.forEach(function(item) {
```
Comment (sebastianbenz) [inline tasks/validate-example.js]: idea: we could use gutils.colors.red to highlight errors (are there warnings as well?).

File: tasks/validate-example.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    if (file.isStream()) {
+      this.emit('error', new PluginError('validate-example',
+            'Streams not supported!'));
+    } else if (file.isBuffer()) {
+
+
+      // write file to disk, invoke validator, capture output & cleanup
+      const inputFilename = path.basename(file.path);
+      const tmpFile = os.tmpdir() + '/' + inputFilename;
+      fs.writeFile(tmpFile, file.contents, 'utf8', function(err) {
```
Comment (sebastianbenz) [inline tasks/validate-example.js]: should we use the passed encoding instead of explicitly using utf8?

File: gulpfile.js (hunk 1 above)
Comment (jkingyens) [inline gulpfile.js]: The validate task doesn't perform a build or touch the output directory. It just tests that the code in "src" will produce valid amp. If we can get validate to run fast (~ 1s) then we could insert it into the build.

Comment (sebastianbenz) [inline gulpfile.js]: What I meant was running validate after build:example, e.g.:

Comment (jkingyens) [inline gulpfile.js]: How do we know which files in dist need to be validated?

Comment (sebastianbenz) [inline gulpfile.js]: All AMP HTML files...

...but we should leave this for another PR.

**Comment by sebastianbenz** [comment]
LGTM
